components/org.wso2.carbon.identity.api.server.script.library/org.wso2.carbon.identity.api.server.script.library.v1/src/main/java/org/wso2/carbon/identity/api/server/script/library/v1/impl/ScriptLibrariesApiServiceImpl.java

```diff
@@ -46,7 +47,8 @@ public Response addScriptLibrary(String name, InputStream contentInputStream, At
 
         serverScriptLibrariesService.addScriptLibrary(name, contentInputStream, description);
         URI location =
-                ContextLoader.buildURIForHeader(V1_API_PATH_COMPONENT + SCRIPT_LIBRARY_PATH_COMPONENT + "/" + name);
+                ContextLoader.buildURIForHeader(
+                        V1_API_PATH_COMPONENT + SCRIPT_LIBRARY_PATH_COMPONENT + "/" + URLEncoder.encode(name));
```

Comment:
```suggestion
                        V1_API_PATH_COMPONENT + SCRIPT_LIBRARY_PATH_COMPONENT + "/" + URLEncoder.encode(name), StandardCharsets.UTF_8.name());              
```

Reply:
also, replace `+` by `%20`, because this reference will be returned in the headers

Reply:
Fixed in f72b9a5